src/libraries/System.Net.Primitives/src/System/Net/IPAddressParser.cs

```diff
@@ -16,7 +18,8 @@ internal class IPAddressParser
     {
         private const int MaxIPv4StringLength = 15; // 4 numbers separated by 3 periods, with up to 3 digits per number
 
-        internal static IPAddress Parse(ReadOnlySpan<char> ipSpan, bool tryParse)
+        [return: NotNullIfNotNull("tryParse")]
```

Comment:
The `tryParse` argument is boolean, so I don't see how it could be null. We don't have an attribute that supports the precondition you have in mind, so it's probably best to just remove it.

Reply:
:-) Fixed.